test/default-xrp-client-test.ts

```diff
@@ -461,4 +469,130 @@ describe('Default Xpring Client', function(): void {
+    //
+    //        const expectedPaymentHistory = [XRPTransaction.from(testPaymentTransaction), XRPTransaction.from(testPaymentTransaction)]
+    // It leaves less room for error in the logic, but it no longer automatically reflects the composition of
+    // the testGetAccountTransactionHistory object
+    const testTransactionResponseList = testGetAccountTransactionHistoryResponse.getTransactionsList()
+    const expectedPaymentHistory: Array<XRPTransaction> = []
+    // eslint-disable-next-line no-restricted-syntax
+    // eslint-disable-next-line no-underscore-dangle
+    for (const _getTransactionResponse of testTransactionResponseList) {
+      const transaction = _getTransactionResponse.getTransaction()
+      switch (transaction?.getTransactionDataCase()) {
+        case Transaction.TransactionDataCase.PAYMENT: {
+          const xrpTransaction = XRPTransaction.from(transaction)
+          if (!xrpTransaction) {
+            throw new Error(XRPClientErrorMessages.paymentConversionFailure)
+          } else {
+            expectedPaymentHistory.push(xrpTransaction)
+          }
+          break
+        }
+        default:
+      }
+    } // end for
+    // THEN the payment history is returned as expected
+    assert.deepEqual(expectedPaymentHistory, paymentHistory)
+  })
+
+  it('Payment History - classic address', function(done): void {
+    // GIVEN an XRPClient and a classic address
+    const xrpClient = new DefaultXRPClient(fakeSucceedingNetworkClient)
+    const classicAddress = 'rsegqrgSP8XmhCYwL9enkZ9BNDNawfPZnn'
+
+    // WHEN the payment history for an account is requested THEN an error to use X-Addresses is thrown.
+    xrpClient.paymentHistory(classicAddress).catch((error) => {
+      assert.typeOf(error, 'Error')
+      assert.equal(error.message, XRPClientErrorMessages.xAddressRequired)
+      done()
+    })
+  })
+
+  it('Payment History - network failure', function(done): void {
+    // GIVEN an XRPClient which wraps an erroring network client.
+    const xrpClient = new DefaultXRPClient(fakeErroringNetworkClient)
+
+    // WHEN the payment history is requested THEN an error is propagated.
+    xrpClient.paymentHistory(testAddress).catch((error) => {
+      assert.typeOf(error, 'Error')
+      assert.equal(error, FakeNetworkClientResponses.defaultError)
+      done()
+    })
+  })
+
+  it('Payment History - non-payment transactions', async function(): Promise<
+    void
+  > {
+    // GIVEN an XRPClient client which will return a transaction history which contains non-payment transactions
+
+    // Generate expected transactions from the default response, which only contains payments.
+    const nonPaymentTransactionResponse = new GetTransactionResponse()
+    nonPaymentTransactionResponse.setTransaction(testCheckCashTransaction)
+    const heteregeneousTransactionHistory = testGetAccountTransactionHistoryResponse // here, homogeneous w/ PAYMENT txns
```

Comment:
I find these comments. a bit ard to follow. I would just write "// transaction history which contains payment and non-payment transactions." as a one liner above the `const hetergenousTransactionHistory`. 

I think inline is harder to read. YMMV and if you feel strongly you can leave it this way.

Reply:
my feelings are nonexistent